src/Select.tsx

```diff
@@ -15,14 +15,23 @@ export type SelectProps = {
         React.SelectHTMLAttributes<HTMLSelectElement>,
         HTMLSelectElement
     >;
-    children: ReactNode;
     /** Default: false */
     disabled?: boolean;
     /** Default: "default" */
     state?: "success" | "error" | "default";
     /** The message won't be displayed if state is "default" */
     stateRelatedMessage?: ReactNode;
     style?: CSSProperties;
+    options: SelectOption[];
+    placeholder?: string;
+};
+
+export type SelectOption<T = string> = {
+    value: T | number | readonly string[] | undefined;
+    label: string;
+    disabled?: boolean;
+    hidden?: boolean;
+    // selected should not be handled directly by option, but as Select's value
```

Comment:
If the component is controlled yes but we want to enable it to be uncontrolled as well.